File: benchmarks/solutions/elixir/r2-json-transform.exs
input = IO.read(:stdio, :eof)

case Jason.decode(input) do
  {:error, _} ->
    IO.puts(:stderr, "error: invalid JSON")
    System.halt(1)

  {:ok, users} ->
    result =
      users
      |> Enum.filter(&(&1["active"] and &1["age"] > 18))
      |> Enum.map(&Map.take(&1, ["name", "email"]))

    IO.puts(Jason.encode!(result))
end

File: benchmarks/solutions/swift/r2-json-transform.swift
import Foundation

struct User: Decodable {
    let name: String
    let age: Int
    let active: Bool
    let email: String
}

struct Output: Encodable {
    let name: String
    let email: String
}

let input = FileHandle.standardInput.readDataToEndOfFile()
guard let users = try? JSONDecoder().decode([User].self, from: input) else {
    fputs("error: invalid JSON\n", stderr)
    exit(1)
}

let result = users
    .filter { $0.active && $0.age > 18 }
    .map { Output(name: $0.name, email: $0.email) }

let output = try! JSONEncoder().encode(result)
print(String(data: output, encoding: .utf8)!)
